### DISCUSSION TITLE
Formatted fields should be displayed as separate boxes like in 6

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5642] [Feature Requests]
Catalog Number formatting currently works much differently in Specify 7 than in Specify 6.

## **Specify 6:**

**You can see that the ABC, numeric, and separators are not meant to be editable.**

https://user-images.githubusercontent.com/37256050/189776304-e867598f-c4b6-4ca4-b3a8-62031755f62b.mov

## **Specify 7:**

Specify 7 does not enforce this configuration at all. This leads to the data entry person not understanding the expected formatting:

https://user-images.githubusercontent.com/37256050/189776420-9d919208-f23c-480c-83ba-0ebb2a2974c7.mov

This applies to any formatted field's formatter. 

---

You can see this for yourself on the `catNum` database on the test server.

https://catnum-latest.test.specifysystems.org/specify/

**Comment by grantfitzsimmons** [comment]
<img width="531" alt="image" src="https://user-images.githubusercontent.com/37256050/206736576-6de9570e-14c2-4f65-9602-dbf126793f1c.png">

This continues to cause issues. I don't even know what the requested formatter is!

**Comment by maxpatiiuk** [comment]
If you remove the value from the field, the placeholder would show the expected format - not ideal, but that's what we have until this is fixed properly

**Comment by grantfitzsimmons** [comment]
https://user-images.githubusercontent.com/37256050/206740349-47c66ced-fd2d-42f2-b6aa-c5a7d4669f08.mov

Sadly this gives me no indication at all

https://dukebryo-testability.test.specifysystems.org/specify/view/accession/new/

Same on `edge` and `testability` so nothing new

**Comment by maxpatiiuk** [comment]
For future reference:
I think this happens only for UiFormatters that don't have autonumbering enabled.
For example, here is the AccessionNumber UI Formatter definition:
```xml
<format system="true" name="AccessionStringFormatter" class="edu.ku.brc.specify.datamodel.Accession" fieldname="accessionNumber" default="true">
    <field type="alphanumeric" size="10" value="AAAAAAAAAA"/>
</format>
```

**Comment by maxpatiiuk** [comment]
This can be fixed alongside #3402. See https://github.com/specify/specify7/issues/3402#issuecomment-1521903523 for more details.

I.e, implement a customizable react component that wll facilitate drawing multiple boxes and managing focus/cursor/backspace logic and use that component both for UIFormatted fields and date fields

**Comment by maxpatiiuk** [comment]
If/when this gets implemented, consider also adding support for `minsize` UI field formatter attribute like in Specify 6:
https://discourse.specifysoftware.org/t/add-regular-expression-format-in-specify-6/560#:~:text=Replace%20the%20Y%20with%20the%20minimum%20value%20your%20catalog%20number%20might%20be%20including%20separators.%20So%20for%20example%20if%20your%20catalog%20number%20might%20be%20as%20short%20as%20%E2%80%9CEX%201234%2D56%E2%80%9D%2C%20you%20would%20enter%2010.

**Comment by grantfitzsimmons** [comment]
>#2 Specimen number editing: so am I correct that since we have to manually enter part of the specimen number to start with, there is no way to have it locked at any point? Would there be any way to revise the database so that the “Specimen number” is part of its own table or something? How was it achieved in our Specify 6 implementation?
(The part we have been manually entering is a number associated with the family of the taxon… would there be any way possibly to automate that so that it is fixed based on the “Family” picklist or the taxon selected for the determination? I realize this could possibly be a much bigger change to the database’s structure, I’m just trying to come up with a way where we don’t accidentally delete or alter this very important number.)

From Taran Meyer at CFIA
